static/src/js/actions/exportSearch.js

```diff
@@ -67,45 +97,68 @@ export const exportSearch = (format) => (dispatch, getState) => {
       pageNum: undefined,
       pageSize: undefined
     }
-  }, format)
-    .then((response) => {
-      const { data } = response
-
-      // Create a blob with the text data from the export
-      let blob
-      if (format === 'csv') {
-        blob = new Blob([data], { type: 'text/csv' })
+  }, format, searchExportParams)
+    .then(async (response) => {
+      // key is a random uuid for this specific export request
+      // we use it to poll the status of the export later
+      const { key } = response.data
+      if (!key) throw Error('server did not respond with a uuid for the export request, which we need in order to find the download')
+
+      const { exportStatusRefreshTime = 5000 } = getApplicationConfig()
+
+      const [error, signedUrl] = await new Promise((resolve) => {
+        const intervalId = setInterval(async () => {
```

Comment:
I'm concerned about this UX, is there anything telling the user the export is processing? What happens if the user changes pages?

Reply:
I agree with Matthew that this UX should be improved, though I don't know that you've really introduced any new behavior. Maybe it could happen in another ticket? Either way, I think there is room for improvement.

Additionally, I dont really love the random characters appended to the exported file. Whats the purpose of that? Does it need to be surfaced to the user? Can we maybe override the filename that appears by default in the save dialog?

Reply:
I removed the random characters append to the end of the exported filename via https://github.com/nasa/earthdata-search/pull/1630/commits/d0eb2b09744b6b60326184b258a980d57e8427c8

Reply:
I originally added the key partial to improve debuggability.  In theory, if a user had an issue with an export, they could provide the filename and that's all we would need to look up the export request.  I'll leave the uuid partial out for now, but if we want to revisit this or if you want to change the filename later, that can definitely happen.

Reply:
I still think we need to look at the UX for this. I can't get the branch to run locally so I haven't been able to try to break an export, but I hate the idea of a user sitting around for a couple of minutes without knowing something is happening